src/js/form.js

```diff
@@ -114,11 +114,9 @@ Form.prototype = {
         const { evaluationCascadeAdditions } = this;
 
         if (evaluationCascadeAdditions.length > 0) {
-            baseEvaluationCascade.push(() => {
-                for (const fn of evaluationCascadeAdditions) {
-                    fn();
-                }
-            });
+            for (const fn of evaluationCascadeAdditions) {
+                baseEvaluationCascade.push(fn);
+            }
```

Comment:
This is the minimal fix. I actually prefer to go back to concatenating the additions as before, but am not sure if there are reasons to use this loop instead. See alternative PR here: https://github.com/enketo/enketo-core/pull/893